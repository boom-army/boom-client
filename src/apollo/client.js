import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
