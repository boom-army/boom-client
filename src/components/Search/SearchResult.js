import React, { useState } from "react";
import SearchResultTweets from "./SearchResultTweets";
import SearchResultTags from "./SearchResultTags";
import SearchResultUsers from "./SearchResultUsers";
import { styled } from "@mui/material/styles";

const Wrapper = styled("div")((props) => ({
  ".tabs": {
    display: "flex",
    justifyContent: "space-around",
    borderBottom: `2px solid ${props.theme.tertiaryColor}`,
  },

  span: {
    cursor: "pointer",
    marginBottom: "0.4rem",
  },

  "span.active": {
    borderBottom: `2px solid ${props.theme.accentColor}`,
    fontWeight: "500",
    color: props.theme.accentColor,
  },
}));

const SearchResult = ({
  searchTweetLoading,
  searchUserLoading,
  searchTagLoading,
  tags,
  users,
  tweets,
}) => {
  const [searchResultAction, setSearchResultAction] = useState("TWEETS");

  const changeToTweets = () => setSearchResultAction("TWEETS");
  const changeToTags = () => setSearchResultAction("TAGS");
  const changeToUsers = () => setSearchResultAction("USERS");

  return (
    <Wrapper>
      <div className="tabs">
        <span
          className={searchResultAction === "TWEETS" ? "active" : ""}
          onClick={changeToTweets}
        >
          Tweets
        </span>
        <span
          className={searchResultAction === "TAGS" ? "active" : ""}
          onClick={changeToTags}
        >
          Tags
        </span>
        <span
          className={searchResultAction === "USERS" ? "active" : ""}
          onClick={changeToUsers}
        >
          Users
        </span>
      </div>

      {searchResultAction === "TWEETS" && (
        <SearchResultTweets tweets={tweets} loading={searchTweetLoading} />
      )}
      {searchResultAction === "TAGS" && (
        <SearchResultTags tags={tags} loading={searchTagLoading} />
      )}
      {searchResultAction === "USERS" && (
        <SearchResultUsers users={users} loading={searchUserLoading} />
      )}
    </Wrapper>
  );
};

export default SearchResult;
