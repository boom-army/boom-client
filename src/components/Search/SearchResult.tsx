import React from "react";
import SearchResultTweets from "./SearchResultTweets";
import SearchResultUsers from "./SearchResultUsers";
import { Box, Tab } from "@mui/material";
import { TabContext, TabList, TabPanel } from "@mui/lab";
import {
  SearchTweetsQuery,
  SearchUserQuery,
} from "../../generated/graphql";

interface SearchResultProps {
  loading: boolean
  searchTweetData: SearchTweetsQuery | undefined
  searchUserData: SearchUserQuery | undefined
  fetchMoreTweets: any
  fetchMoreUsers: any
  tabValue: string
  setTabValue: React.Dispatch<React.SetStateAction<string>>
}

const SearchResult = ({
  loading,
  searchTweetData,
  searchUserData,
  fetchMoreTweets,
  fetchMoreUsers,
  tabValue,
  setTabValue,
}: SearchResultProps) => {
  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
    setTabValue(newValue);
  };

  return (
    <>
      <Box sx={{ typography: "body1" }}>
        <TabContext value={tabValue}>
          <Box sx={{ borderBottom: 2, borderColor: "divider" }}>
            <TabList
              onChange={handleChange}
              aria-label="Search tablist select"
              variant="fullWidth"
            >
              <Tab sx={{ minWidth: 150 }} label="Meeps" value="MEEPS" />
              <Tab label="Tags" value="TAGS" />
              <Tab label="Users" value="USERS" />
            </TabList>
          </Box>
          <TabPanel value="MEEPS">
            <SearchResultTweets tweets={searchTweetData} loading={loading} fetchMoreTweets={fetchMoreTweets} />
          </TabPanel>
          <TabPanel value="TAGS">
            <SearchResultTweets tweets={searchTweetData} loading={loading} fetchMoreTweets={fetchMoreTweets} />
          </TabPanel>
          <TabPanel value="USERS">
            <SearchResultUsers users={searchUserData} loading={loading} fetchMoreUsers={fetchMoreUsers} />
          </TabPanel>
        </TabContext>
      </Box>
    </>
  );
};

export default SearchResult;
