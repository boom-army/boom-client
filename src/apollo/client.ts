import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
