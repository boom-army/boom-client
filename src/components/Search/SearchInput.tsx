import React from "react";
import SearchResult from "./SearchResult";
import { useInput } from "../../hooks/useInput";
import {
  SEARCH_BY_TAG,
  SEARCH_BY_TWEET,
  SEARCH_BY_USER,
} from "../../queries/search";
import { displayError } from "../../utils";
import { useLazyQuery } from "@apollo/client";
import { useSnackbar } from "notistack";
import { styled } from "@mui/material/styles";

const Wrapper = styled("div")((props) => ({
  margin: "1rem 0",
  marginLeft: "1rem",
  input: {
    height: "40px",
    width: "70%",
    borderRadius: "30px",
    background: props.theme.tertiaryColor2,
    border: props.theme.tertiaryColor2,
    color: props.theme.secondaryColor,
    fontFamily: props.theme.font,
    fontSize: "1rem",
    paddingLeft: "1.2rem",
  },

  "@media screen and (max-width: 530px)": {
    input: {
      fontSize: "0.9rem",
    },
  },
}));

const SearchInput = () => {
  const term = useInput("");

  const [searchByTag, { data: searchTagData, loading: searchTagLoading }] =
    useLazyQuery(SEARCH_BY_TAG);

  const [
    searchByTweet,
    { data: searchTweetData, loading: searchTweetLoading },
  ] = useLazyQuery(SEARCH_BY_TWEET);

  const [searchByUser, { data: searchUserData, loading: searchUserLoading }] =
    useLazyQuery(SEARCH_BY_USER);

  const { enqueueSnackbar } = useSnackbar();

  const handleSearch = async (e: any) => {
    e.preventDefault();

    if (!term.value) {
      return enqueueSnackbar("Enter something to search", { variant: "error" });
    }

    try {
      await searchByTag({ variables: { term: term.value } });
      await searchByTweet({ variables: { term: term.value } });
      await searchByUser({ variables: { term: term.value } });
    } catch (err) {
      displayError(err, enqueueSnackbar);
    }
    term.setValue("");
  };

  return (
    <>
      <Wrapper>
        <form onSubmit={(e) => handleSearch(e)}>
          <input
            placeholder="Search by tags, tweets, people"
            type="text"
            value={term.value}
            onChange={term.onChange}
          />
        </form>
      </Wrapper>
      <SearchResult
        searchTagLoading={searchTagLoading}
        searchTweetLoading={searchTweetLoading}
        searchUserLoading={searchUserLoading}
        tags={searchTagData}
        users={searchUserData}
        tweets={searchTweetData}
      />
    </>
  );
};

export default SearchInput;
