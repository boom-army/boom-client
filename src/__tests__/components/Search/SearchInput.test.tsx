import React from "react";
import renderer from "react-test-renderer";
import {
  SEARCH_BY_TAG,
  SEARCH_BY_TWEET,
  SEARCH_BY_USER,
} from "../../../queries/search";

import SearchInput from "../../../components/Search/SearchInput";

import { User as UserProps } from "../../../generated/graphql";
import SearchResult from "../../../components/Search/SearchResult";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SnackbarProvider } from "notistack";
import { MockedProvider } from "@apollo/client/testing";
import { createSerializer } from '@emotion/jest'

expect.addSnapshotSerializer(createSerializer(
  {
    classNameReplacer(className, index) {
      return `css-${index}`
    },
       // includeStyles: false 
  }
));


export const __mocks__: any = [
  {
    request: {
      query: SEARCH_BY_TAG,
      variables: {
        term: {
          value: "",
        },
      },
    },
    result: {
      data: {},
      errors: "An error occurred",
    },
  },
];

describe("Search Input:", () => {
  test("display <SearchInput/> component when users undefined", () => {
    const tree = render(
      <MockedProvider mocks={__mocks__} addTypename={false}>
        <SnackbarProvider>
          <SearchInput />
        </SnackbarProvider>
      </MockedProvider>,
      { wrapper: MemoryRouter }
    );
    // fireEvent.click(tree.getByText("Users"));
    expect(tree).toMatchSnapshot();
  });
});

test("Assertion testing of <SearchInput/> component", () => {
  render(
    <MockedProvider mocks={__mocks__} addTypename={false}>
      <SnackbarProvider>
        <SearchInput />
      </SnackbarProvider>
    </MockedProvider>,
    { wrapper: MemoryRouter }
  );

  expect(screen.getByRole("textbox", { name: "" })).toHaveAttribute(
    "placeholder",
    "Search by tags, tweets, people"
  );
});
