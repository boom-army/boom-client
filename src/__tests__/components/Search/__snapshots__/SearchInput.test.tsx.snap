// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Input: display <SearchInput/> component when users undefined 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1n1p729"
      >
        <form>
          <input
            placeholder="Search by tags, tweets, people"
            type="text"
            value=""
          />
        </form>
      </div>
      <div
        class="css-1mw65dx"
      >
        <div
          class="tabs"
        >
          <span
            class="active"
          >
            Tweets
          </span>
          <span
            class=""
          >
            Tags
          </span>
          <span
            class=""
          >
            Users
          </span>
        </div>
        <div
          class="css-15inajd"
        >
          <p>
            Use the search bar to find tags, people and tweets
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1n1p729"
    >
      <form>
        <input
          placeholder="Search by tags, tweets, people"
          type="text"
          value=""
        />
      </form>
    </div>
    <div
      class="css-1mw65dx"
    >
      <div
        class="tabs"
      >
        <span
          class="active"
        >
          Tweets
        </span>
        <span
          class=""
        >
          Tags
        </span>
        <span
          class=""
        >
          Users
        </span>
      </div>
      <div
        class="css-15inajd"
      >
        <p>
          Use the search bar to find tags, people and tweets
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
